testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/page/ApplicationsPage.java

```diff
@@ -17,14 +17,46 @@
 
 package org.keycloak.testsuite.ui.account2.page;
 
+import org.keycloak.representations.account.ClientRepresentation;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * @author Vaclav Muzikar <vmuzikar@redhat.com>
  */
 public class ApplicationsPage extends AbstractLoggedInPage {
+    @FindBy(xpath = "//ul[@id='applications-list']/li/div/div[@class='pf-c-data-list__item-content']")
```

Comment:
I'm not sure we should use selectors like that. This will be very error prone in the future when slightly changing the layout or even updating PF/React. Please add IDs where missing. (To keep all IDs unique, you can make a part of the ID dynamic like e.g. [here](https://github.com/keycloak/keycloak/blob/210fd92d2328b3d8173d7c07e2708e6d9544a396/themes/src/main/resources/theme/keycloak-preview/account/resources/app/content/device-activity-page/DeviceActivityPage.tsx#L283)).